float64 x
float64 y
int32 hash
float64 error